### PR TITLE
Do not use VLAs in vec256_qint.h

### PR DESCRIPTION
**Pull request by malfet**
Use `std::decay_t<decltype(foo)>::size()`  instead of `foo.size()` to help compiler with static array allocations.
Even if `Vec256::size()` is `constexpr`, `foo.size()` (where `foo` is of type `Vec256`) is not an integral constant expression, therefore compiler have to use VLAs, which are not part of C++ standard.

Test Plan: CI

### CODE REVIEW

File: aten/src/ATen/cpu/vec256/vec256_qint.h [hunk 1]
```diff
@@ -413,12 +413,12 @@ Vec256<c10::qint32> inline operator*(
   return _mm256_mullo_epi32(a, b);
 #else
   // Pray the compiler can autovectorize this
-  int32_t a_vals[a.size()];
-  int32_t b_vals[b.size()];
+  int32_t a_vals[std::decay_t<decltype(a)>::size()];
```
Comment (malfet) [inline aten/src/ATen/cpu/vec256/vec256_qint.h:416]: I wonder if it's better to use `std::array` here, to avoid any `constexpr` interpretation ambiguity.

Comment (dzhulgakov) [reply aten/src/ATen/cpu/vec256/vec256_qint.h:416]: Great catch! Imho `std:array` here would work just fine

Comment (EscapeZero) [reply aten/src/ATen/cpu/vec256/vec256_qint.h:416]: I second the `std::array` proposal here for sure.

**Comment by malfet** [comment]
Added 2nd commit that replaces `int32_t foo[size()];` arrays with `std::array<int32_t, size()> foo;`

**Review by dzhulgakov — APPROVED** [review]

**Pull request closed without merging**